src/main/java/de/tum/in/www1/artemis/web/rest/AttachmentResource.java

```diff
@@ -65,41 +66,50 @@ public AttachmentResource(AttachmentRepository attachmentRepository, GroupNotifi
     /**
      * POST /attachments : Create a new attachment.
      *
-     * @param attachment the attachment to create
+     * @param attachment the attachment object to create
+     * @param file       the file to save
      * @return the ResponseEntity with status 201 (Created) and with body the new attachment, or with status 400 (Bad Request) if the attachment has already an ID
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
-    @PostMapping("/attachments")
+    @PostMapping(value = "attachments", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     @PreAuthorize("hasRole('EDITOR')")
-    public ResponseEntity<Attachment> createAttachment(@RequestBody Attachment attachment) throws URISyntaxException {
+    public ResponseEntity<Attachment> createAttachment(@RequestPart Attachment attachment, @RequestPart MultipartFile file) throws URISyntaxException {
         log.debug("REST request to save Attachment : {}", attachment);
-        if (attachment.getId() != null) {
-            throw new BadRequestAlertException("A new attachment cannot already have an ID", ENTITY_NAME, "idExists");
-        }
+        attachment.setId(null);
+
+        String pathString = fileService.handleSaveFile(file, false, false);
+        attachment.setLink(pathString);
+
         Attachment result = attachmentRepository.save(attachment);
         this.cacheManager.getCache("files").evict(fileService.actualPathForPublicPath(result.getLink()));
         return ResponseEntity.created(new URI("/api/attachments/" + result.getId())).body(result);
     }
 
     /**
-     * PUT /attachments : Updates an existing attachment.
+     * PUT /attachments/:id : Updates an existing attachment.
      *
+     * @param id               the id of the attachment to save
      * @param attachment       the attachment to update
-     * @param notificationText text that will be send to student group
+     * @param file             the file to save if the file got changed (optional)
+     * @param notificationText text that will be sent to student group
      * @return the ResponseEntity with status 200 (OK) and with body the updated attachment, or with status 400 (Bad Request) if the attachment is not valid, or with status 500
      *         (Internal Server Error) if the attachment couldn't be updated
      */
-    @PutMapping("/attachments")
+    @PutMapping(value = "attachments/{id}", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     @PreAuthorize("hasRole('EDITOR')")
-    public ResponseEntity<Attachment> updateAttachment(@RequestBody Attachment attachment, @RequestParam(value = "notificationText", required = false) String notificationText) {
+    public ResponseEntity<Attachment> updateAttachment(@PathVariable Long id, @RequestPart Attachment attachment, @RequestPart(required = false) MultipartFile file,
```

Comment:
How about using `attachmentId`. This makes it clearer what the variable refers to.
We also do this for the exercise and course endpoints and probably many more